packages/apps/virtual-machine/values.yaml

```diff
@@ -23,6 +24,7 @@ resources:
   disk: 5Gi
 sshPwauth: true
 disableRoot: true
+user: username
 password: hackme
 chpasswdExpire: false
```

Comment:
What do you think about using common `cloudInit` config parameter for this?

Reply:
It might be better to pass all cloud-init configuration as a string:

```yaml
cloudInitUserConfig: |
  #cloud-config
  ssh_pwauth: true
  disable_root: true
  user: username
```

due to fact user configuration have no formalized spec, it can include multiple header, scripts and even jinja2 templates
https://cloudinit.readthedocs.io/en/latest/explanation/instancedata.html#example-cloud-config-with-instance-data

variuos vendors can specify their own format for this, eg. Talos Linux allows to write machine-config in user-data

This config should be saved into external secret and included with secretRef into KubeVirt VM:
```yaml
    - name: cloudinitdisk
      cloudInitNoCloud:
        secretRef:
          name: my-vmi-secret
```

As about sshKeys, cloud-init defines formalized spec for this in meta-data:
https://cloudinit.readthedocs.io/en/latest/explanation/instancedata.html#v1-public-ssh-keys

KubeVirt allows to specify them as 

```yaml
      accessCredentials:
      - sshPublicKey:
          source:
            secret:
              secretName: my-pub-key
          propagationMethod:
            noCloud: {}
```

details:
- https://kubevirt.io/user-guide/user_workloads/startup_scripts/
- https://kubevirt.io/user-guide/user_workloads/accessing_virtual_machines/